torch/_inductor/codegen/triton.py

```diff
@@ -850,11 +850,9 @@ def floordiv(a, b):
 
     @staticmethod
     def sign(x):
-        def to_int(s):
-            return f"{s}.to(tl.int8)"
-
-        left = to_int(ops.lt("0", x))
-        right = to_int(ops.lt(x, "0"))
+        z = ops.constant(0, torch.int32)
+        left = ops.to_dtype((ops.lt(z, x)), torch.int8)
+        right = ops.to_dtype((ops.lt(x, z)), torch.int8)
         sub = ops.sub(left, right)
         return f"{sub}.to({x}.dtype)"
```

Comment:
What's the issue that requires all these to be updated. I'm not opposed, just curious.

Reply:
With this approach, either all the lowerings are written with strings or all of them are written with ops.If you mix them, you end up getting strings as the inputs in the bound propagation and we don't know how to "bound these strings". You can sidestep this limitation using CSE variables and setting the bounds properly on them, but yeah.

Reply:
Let me write a comment in the part where we forward the bounds.